addons/binding/org.openhab.binding.nibeuplink/src/main/java/org/openhab/binding/nibeuplink/handler/UplinkBaseHandler.java

```diff
@@ -0,0 +1,187 @@
+import org.openhab.binding.nibeuplink.internal.command.UpdateSetting;
+import org.openhab.binding.nibeuplink.internal.connector.UplinkWebInterface;
+import org.openhab.binding.nibeuplink.internal.model.Channel;
+import org.openhab.binding.nibeuplink.internal.model.CustomChannels;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link UplinkBaseHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Friese - initial contribution
+ */
+@NonNullByDefault
+public abstract class UplinkBaseHandler extends BaseThingHandler implements NibeUplinkHandler, AtomicReferenceUtils {
+    private final Logger logger = LoggerFactory.getLogger(UplinkBaseHandler.class);
+
+    private Set<Channel> deadChannels = new HashSet<>(100);
+
+    /**
+     * Refresh interval which is used to poll values from the NibeUplink web interface (optional, defaults to 60 s)
+     */
+    private int refreshInterval;
+
+    /**
+     * Refresh interval which is used clean the dead channel list (optional, defaults to 1 h)
+     */
+    private int houseKeepingInterval = 1;
+
+    /**
+     * Interface object for querying the FRITZ!Box web interface
+     */
+    private UplinkWebInterface webInterface;
+
+    /**
+     * Schedule for polling
+     */
+    private final AtomicReference<@Nullable Future<?>> pollingJobReference;
+
+    /**
+     * Schedule for periodic cleaning dead channel list
+     */
+    private final AtomicReference<@Nullable Future<?>> deadChannelHouseKeepingReference;
+
+    public UplinkBaseHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.webInterface = new UplinkWebInterface(getConfiguration(), scheduler, this, httpClient);
+        this.pollingJobReference = new AtomicReference<@Nullable Future<?>>(null);
+        this.deadChannelHouseKeepingReference = new AtomicReference<@Nullable Future<?>>(null);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!(command instanceof RefreshType)) {
+            logger.debug("command for {}: {}", channelUID.getIdWithoutGroup(), command.toString());
+            Channel channel = getSpecificChannel(channelUID.getIdWithoutGroup());
+            if (channel != null && !channel.isReadOnly()) {
+                webInterface.enqueueCommand(new UpdateSetting(this, channel, command));
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("About to initialize NibeUplink");
+        NibeUplinkConfiguration config = getConfiguration();
+
+        logger.debug("NibeUplink initialized with configuration: {}", config);
+
+        setupCustomChannels(config);
+        this.refreshInterval = config.getPollingInterval();
+        this.houseKeepingInterval = config.getHouseKeepingInterval();
+
+        startPolling();
+        webInterface.start();
```

Comment:
I thought the initial status is "initializing" and that would be correct even if the startup takes 30 seconds. It does not block it returns immediately. But if you prefer "Unknown" I could set this.

Please confirm.

Reply:
Right, initial status is `INITIALIZING` which is set by the framework. The framework expects however that the thing handler sets a new status immediately, takes over responsibility and returns the initializing thread. In case you can not determine online/offline status right away `UNKOWN` is to be used.
Your long running initialization is then run by another thread pool which is exclusive for thing handers.

Reply:
Ok, status will now be updated immediately.